Add a few things to make installing and testing easier

bd808 opened the pull request:
- Add requirements.txt files for easy pip installs
- Add tox configuration to make running py.test, flake8 and sphinx easy
- Fix flake8 errors and warnings
- Move comments into docblocks to improve generated documentation

Supersedes pull/#1

bd808 committed:
Add requirements files

Declare requirements for easy installation with pip.

bd808 committed:
Add tox support

Add configuration to run py.test, flake8 and sphinx with tox.

bd808 committed:
Fix errors and warnings from tox tests

* Fix various errors and warnings from pep8 checks.
* Move comments to docblocks describing methods and objects.

robla commented on wbstatus.py:
```diff
@@ -106,18 +107,20 @@ def get_user(self, phid):
         self.tasknums = tasknums
 
     def load_from_phabricator(self, phab, cachedir):
-        taskquerycall = lambda: phab.maniphest.query(ids=self.tasknums)
```
The changes from using lambdas to named functions breaks the script.  I haven't had a chance to debug it fully, but I suspect it's because the scope of "self" is different when the functions are assigned via anonymous function, and that declaring them this way screws up the closure.  The error I'm getting is:

Traceback (most recent call last):
  File "./wbstatus.py", line 512, in <module>
    main()
  File "./wbstatus.py", line 432, in main
    for tasknum, taskfeed in activity.iteritems():
AttributeError: 'NoneType' object has no attribute 'iteritems'

bd808 replied:
Changed back to lambdas in the next patch. I haven't actually run this successfully locally do to a lack of workboard html scrapes.

bd808 committed:
Convert nested defs back to lambdas

Rather than using nested defs which mangle `self` in the runtime scope,
convert back to using lambdas for callbacks, but declare them inline
rather than assigning to local variables. This makes the pep8 checker
happy.

bd808 committed:
Remove html_static_path from sphinx config

We aren't using it and having the config without making an empty
directory makes sphinx whine.
